aten/src/ATen/native/cpu/SumKernel.cpp

```diff
@@ -95,7 +95,7 @@ struct InnerSumCastLoadPolicy {
   static vacc_t load(const char * C10_RESTRICT data, int64_t stride, int64_t index) {
     auto ptr = reinterpret_cast<const scalar_t*>(data + stride * index);
     return load_reduce_vec<acc_t>(ptr, [](acc_t a, scalar_t b) {
-      return a + b;
+      return a + static_cast<acc_t>(b);
```

Comment:
Template substitution fails with conflicting type `complex<float>` and `complex<Half>`

https://github.com/pytorch/pytorch/blob/2b7943c47c8561a46103488b0fe9a592b87dc5bb/c10/util/complex.h#L339-L342